File: src/main/java/br/com/dusty/dkits/listener/Listeners.java
package br.com.dusty.dkits.listener;

import br.com.dusty.dkits.Main;
import br.com.dusty.dkits.listener.gameplay.PlayerInteractListener;
import br.com.dusty.dkits.listener.login.AsyncPlayerPreLoginListener;
import br.com.dusty.dkits.listener.login.PlayerJoinListener;
import br.com.dusty.dkits.listener.login.PlayerLoginListener;
import br.com.dusty.dkits.listener.mechanics.*;
import br.com.dusty.dkits.listener.quit.PlayerQuitListener;
import org.bukkit.Bukkit;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerCommandPreprocessEvent;
import org.bukkit.plugin.PluginManager;

import java.util.ArrayList;

public class Listeners {
	
	/**
	 * {@link ArrayList} que contém todos os {@link Listener} a serem/já registrados pelo plugin.
	 */
	private static final ArrayList<Listener> LISTENERS = new ArrayList<>();
	
	/**
	 * Registra todos os {@link Listener} da {@link ArrayList} LISTENERS.
	 */
	public static void registerAll() {
		//Usage: LISTENERS.add(new FooListener());
		
		//Login
		LISTENERS.add(new AsyncPlayerPreLoginListener());
		LISTENERS.add(new PlayerJoinListener());
		LISTENERS.add(new PlayerLoginListener());
		
		//Gameplay
		LISTENERS.add(new PlayerInteractListener());
		
		//Mechanincs
		LISTENERS.add(new EntityDamageListener());
		LISTENERS.add(new FoodLevelChangeListener());
		LISTENERS.add(new ItemSpawnListener());
		LISTENERS.add(new LeavesDecayListener());
		LISTENERS.add(new PlayerCommandPreprocessListener());
		LISTENERS.add(new PlayerMoveListener());
		LISTENERS.add(new ServerListPingListener());
		LISTENERS.add(new SignChangeListener());
		LISTENERS.add(new WeatherChangeListener());
		
		//Quit
		LISTENERS.add(new PlayerQuitListener());
		
		PluginManager pluginManager = Bukkit.getPluginManager();
		LISTENERS.forEach(listener -> pluginManager.registerEvents(listener, Main.INSTANCE));
	}
}

File: src/main/kotlin/br/com/dusty/dkits/listener/Listeners.kt
package br.com.dusty.dkits.listener

import br.com.dusty.dkits.Main
import br.com.dusty.dkits.listener.gameplay.EntityDamageByEntityListener
import br.com.dusty.dkits.listener.gameplay.PlayerDeathListener
import br.com.dusty.dkits.listener.gameplay.PlayerInteractListener
import br.com.dusty.dkits.listener.login.AsyncPlayerPreLoginListener
import br.com.dusty.dkits.listener.login.PlayerJoinListener
import br.com.dusty.dkits.listener.login.PlayerLoginListener
import br.com.dusty.dkits.listener.mechanics.*
import br.com.dusty.dkits.listener.quit.PlayerQuitListener
import org.bukkit.Bukkit
import org.bukkit.event.Listener
import java.util.*

object Listeners {

	/**
	 * [ArrayList] que contém todos os [Listener] a serem/já registrados pelo plugin.
	 */
	private val LISTENERS = hashSetOf<Listener>()

	/**
	 * Registra todos os [Listener] da [ArrayList] LISTENERS.
	 */
	fun registerAll() {
		//Usage: LISTENERS.add(FooListener)

		//Login
		LISTENERS.add(AsyncPlayerPreLoginListener)
		LISTENERS.add(PlayerJoinListener)
		LISTENERS.add(PlayerLoginListener)

		//Gameplay
		LISTENERS.add(EntityDamageByEntityListener)
		LISTENERS.add(PlayerDeathListener)
		LISTENERS.add(PlayerInteractListener)

		//Mechanincs
		LISTENERS.add(AsyncPlayerChatListener)
		LISTENERS.add(BlockBreakListener)
		LISTENERS.add(BlockPlaceListener)
		LISTENERS.add(EntityDamageListener)
		LISTENERS.add(EntityPickupItemListener)
		LISTENERS.add(FoodLevelChangeListener)
		LISTENERS.add(InventoryClickListener)
//		LISTENERS.add(ItemSpawnListener)
		LISTENERS.add(PlayerCommandPreProcessListener)
		LISTENERS.add(LeavesDecayListener)
		LISTENERS.add(PlayerDropItemListener)
		LISTENERS.add(PlayerInteractEntityListener)
		LISTENERS.add(PlayerItemDamageListener)
		LISTENERS.add(PlayerMoveListener)
		LISTENERS.add(ServerListPingListener)
		LISTENERS.add(SignChangeListener)
		LISTENERS.add(WeatherChangeListener)
		LISTENERS.add(WorldLoadListener)

		//Quit
		LISTENERS.add(PlayerQuitListener)

		val pluginManager = Bukkit.getPluginManager()
		LISTENERS.forEach { listener -> pluginManager.registerEvents(listener, Main.INSTANCE) }
	}
}
